bundles/org.openhab.ui/web/src/pages/settings/file-definition-mixin.js

```diff
@@ -0,0 +1,97 @@
+import Vue from 'vue'
+import Clipboard from 'v-clipboard'
+
+Vue.use(Clipboard)
+
+export default {
+  data () {
+    return {
+      ObjectType: Object.freeze({
+        THING: 'thing',
+        ITEM: 'item'
+      })
+    }
+  },
+  methods: {
+    /**
+     * Copies the file definitions of the given list of thingUIDs or item names to the clipboard.
+     *
+     * @param {string} type - The type of the objects (`thing` or `item`). Use {ObjectType} enum for clarity.
+     * @param {Array} objectIds - The list of object ids to copy. For Things, this should be an array of Thing UIDs.
+     *                            For Items, this should be an array of Item names.
+     *                            When `null`, all objects of the given type will be copied.
+     */
+    copyFileDefinitionToClipboard (type, objectIds = null) {
+      const typeLabel = type.charAt(0).toUpperCase() + type.slice(1) + 's'
+
+      let copiedObjectsLabel = null
+      if (objectIds === null) {
+        copiedObjectsLabel = `All ${typeLabel}`
+      } else if (objectIds.length === 1) {
+        copiedObjectsLabel = '<b>' + objectIds[0] + '</b>'
+      } else {
+        copiedObjectsLabel = `${objectIds.length} ${typeLabel}`
+      }
+
+      this.$f7.dialog
+        .create({
+          title: `Copy ${typeLabel} File Definition`,
+          text: `Select the file format to copy ${copiedObjectsLabel} to clipboard`,
+          buttons: [
+            {
+              text: 'Cancel',
+              color: 'gray'
+            },
+            {
+              text: 'DSL',
+              color: 'teal',
+              onClick: () => this.executeFileDefinitionCopy(type, typeLabel, objectIds, copiedObjectsLabel, 'DSL', `text/vnd.openhab.dsl.${type}`)
+            },
+            {
+              text: 'YAML',
+              color: 'blue',
+              onClick: () => this.executeFileDefinitionCopy(type, typeLabel, objectIds, copiedObjectsLabel, 'YAML', 'application/yaml')
+            }
+          ]
+        })
+        .open()
+    },
+    // This is a "private" method that is called by the copyFileDefinitionToClipboard method
+    executeFileDefinitionCopy (type, typeLabel, objectIds, copiedObjectsLabel, fileFormatLabel, mediaType) {
```

Comment:
You can try to define this as `function` before defining export default, this way it is really private.

Reply:
how do I have access to `this`? I suppose the caller can pass it?

Reply:
I've moved it out